rs/ic_os/dev_test_tools/setupos-disable-checks/src/main.rs

```diff
@@ -4,91 +4,117 @@ use std::path::{Path, PathBuf};
+    defeat_installer: bool,
+}
+
+fn remove_argument(cmdline: String, argument: String) -> String {
+    let replacements = [
+        (argument.to_string() + "=1"),
+        (argument.to_string() + "=0"),
+        argument,
+    ];
+    let mut existing_args = cmdline.clone();
+    for r in replacements.iter() {
+        existing_args = existing_args.replace(r.as_str(), "");
+        existing_args = existing_args.replace("  ", " ");
+        existing_args = existing_args.trim_matches(' ').to_string();
+    }
+    existing_args
 }
 
 #[tokio::main]
 async fn main() -> Result<(), Error> {
     let cli = Cli::parse();
+    let extra_boot_args_path = Path::new("/extra_boot_args");
+    let extra_boot_args_re = Regex::new(r"EXTRA_BOOT_ARGS=(.*)").unwrap();
 
-    // Open rootfs partition
-    println!("Opening root partition");
-    let mut rootfs = ExtPartition::open(cli.image_path, Some(6)).await?;
+    // Open boot file system.
+    eprintln!("Opening boot file system {}", cli.image_path.display());
+    let mut bootfs = ExtPartition::open(cli.image_path, Some(5)).await?;
 
-    // Overwrite age checks
-    println!("Clearing SetupOS-age checks");
-    let check_setupos_age = NamedTempFile::new()?;
-    fs::write(
-        check_setupos_age.path(),
-        indoc::formatdoc!(
-            r#"
-                #!/usr/bin/env bash
-                echo "Skipping SetupOS-age checks."
-            "#
-        ),
-    )
-    .await
-    .context("failed to write file")?;
-    fs::set_permissions(check_setupos_age.path(), Permissions::from_mode(0o755)).await?;
-    rootfs
-        .write_file(
-            check_setupos_age.path(),
-            Path::new("/opt/ic/bin/check-setupos-age.sh"),
-        )
-        .await?;
+    // Overwrite checks.
+    if cli.defeat_installer {
+        eprintln!(
+            "Defeating installer routine as well as age, hardware and network checks in SetupOS"
+        );
+    } else {
+        eprintln!("Defeating age, hardware and network checks in SetupOS");
+    }
 
-    // Overwrite hardware checks
-    println!("Clearing hardware checks");
-    let hardware = NamedTempFile::new()?;
-    fs::write(
-        hardware.path(),
-        indoc::formatdoc!(
-            r#"
-                #!/usr/bin/env bash
-                echo "Skipping hardware checks."
-            "#
-        ),
-    )
-    .await
-    .context("failed to write file")?;
-    fs::set_permissions(hardware.path(), Permissions::from_mode(0o755)).await?;
-    rootfs
-        .write_file(hardware.path(), Path::new("/opt/ic/bin/check-hardware.sh"))
-        .await?;
+    let mut to_append = "".to_string()
+        + &CHECK_DISABLER_CMDLINE_ARGS
+            .iter()
+            .map(|x| x.to_string() + "=0")
+            .collect::<Vec<String>>()
+            .join(" ");
+    if cli.defeat_installer {
+        to_append = to_append + " " + CHECK_INSTALL_DISABLER_CMDLINE_ARGS[0];
+    }
+
+    let new_args =
```

Comment:
Could you add a few tests to ensure that the string replacements work as expected?

Reply:
Will do today.  After foooooood.  I'm soo hungry.

Reply:
I've pushed a commit d1cbb7192563b23e342d76fbad8d6060d53413ca that adds said tests.